### PR TITLE
CodeExplorerBundle

### PR DESCRIPTION
**Pull request by voronkovich**
Very simple implementation, without any overheads and overkills :)
See https://github.com/symfony/symfony-demo/issues/196

### CODE REVIEW

File: src/ShowSourceBundle/Twig/SourceCodeExtension.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function getFunctions()
 
     public function showSourceCode(\Twig_Environment $twig, \Twig_Template $template)
     {
-        return $twig->render('default/_source_code.html.twig', array(
+        return $twig->render('ShowSourceBundle::_source_code.html.twig', array(
```
Comment (javiereguiluz) [inline src/ShowSourceBundle/Twig/SourceCodeExtension.php]: I'm not sure, but maybe we should rename `_source_code.html.twig` to `source_code.html.twig`. The reason is that from the point of view of the new bundle, this is not a fragment but a "real template" used to display its contents.

**Comment by javiereguiluz** [comment]
I like these changes. My only concern is about the name of the bundle: `ShowSourceBundle` looks too specific to me. I know that this bundle won't grow in the future with new features (at least not with lots of features), but nevertheless I'd prefer a name which is a bit more generic.

File: src/ShowSourceBundle/Twig/SourceCodeExtension.php (hunk 1 above)
Comment (stof) [inline src/ShowSourceBundle/Twig/SourceCodeExtension.php]: I agree with you

**Comment by voronkovich** [comment]
@javiereguiluz, I would suggest the `ShowSourceCodeBundle`. What do you think?

**Comment by voronkovich** [comment]
`CodeExplorerBundle` sounds great for me

File: src/CodeExplorerBundle/Twig/SourceCodeExtension.php [hunk 2]
```diff
@@ -49,7 +49,7 @@ public function getFunctions()
 
     public function showSourceCode(\Twig_Environment $twig, \Twig_Template $template)
     {
-        return $twig->render('default/_source_code.html.twig', array(
+        return $twig->render('CodeExplorerBundle::source_code.html.twig', array(
```
Comment (javiereguiluz) [inline src/CodeExplorerBundle/Twig/SourceCodeExtension.php]: My final comment before merging this change: could/should we use `@CodeExplorer/source_code.html.twig` notation instead?  Symfony still has't made an official statement, but the namespaced syntax seems to be favored for modern apps.

Comment (voronkovich) [inline src/CodeExplorerBundle/Twig/SourceCodeExtension.php]: @javiereguiluz,  I think we should use it here. This notation more clearly for understanding. I like it.

File: src/CodeExplorerBundle/DependencyInjection/CodeExplorerExtension.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeExplorerBundle\DependencyInjection;
+
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader;
```
Comment (xabbuh) [inline src/CodeExplorerBundle/DependencyInjection/CodeExplorerExtension.php]: imo you should import the class name and not the parent namespace here

Comment (voronkovich) [inline src/CodeExplorerBundle/DependencyInjection/CodeExplorerExtension.php]: @xabbuh, thanks! I've fixed this issue.

File: src/CodeExplorerBundle/Resources/config/services.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        class:     CodeExplorerBundle\Twig\SourceCodeExtension
+        arguments: ["@twig.loader", "%kernel.root_dir%"]
+        tags:
+            - { name: twig.extension }
+
+    code_explorer.controller_listener:
+        class:     CodeExplorerBundle\EventListener\ControllerListener
+        arguments: ["@code_explorer.twig.source_code_extension"]
+        tags:
+            - { name: 'kernel.event_listener', event: kernel.controller, method: registerCurrentController }
```
Comment (xabbuh) [inline src/CodeExplorerBundle/Resources/config/services.yml]: single quotes around the argument are not needed

File: src/CodeExplorerBundle/Resources/config/services.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    code_explorer.twig.source_code_extension:
+        public:    false
+        class:     CodeExplorerBundle\Twig\SourceCodeExtension
+        arguments: ["@twig.loader", "%kernel.root_dir%"]
+        tags:
+            - { name: twig.extension }
+
+    code_explorer.controller_listener:
+        class:     CodeExplorerBundle\EventListener\ControllerListener
+        arguments: ["@code_explorer.twig.source_code_extension"]
```
Comment (xabbuh) [inline src/CodeExplorerBundle/Resources/config/services.yml]: double quotes around the argument are not needed here

Comment (voronkovich) [inline src/CodeExplorerBundle/Resources/config/services.yml]: @xabbuh, you're right. I've removed all quotes.

**Comment by javiereguiluz** [comment]
@voronkovich you did a really nice work with this pull request. Sorry it took so long to merge it.

**Pull request closed without merging**